File: frontend/src/models/dto/UserDto.js
export class UserDto {
    /**
     * @name JavaUserDto
     * @class
     * @property id {Number}
     * @property email {String}
     * @property phone {String}
     * @property name {String}
     * @property roleName {String}
     */

    /**
     * @param dto {JavaUserDto}
     */
    constructor(dto) {
        this._id = dto.id;
        this._email = dto.email;
        this._phone = dto.phone;
        this._name = dto.name;
        this._roleName = dto.roleName;
    }

    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get email() {
        return this._email;
    }

    set email(value) {
        this._email = value;
    }

    get phone() {
        return this._phone;
    }

    set phone(value) {
        this._phone = value;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }

    get roleName() {
        return this._roleName;
    }

    set roleName(value) {
        this._roleName = value;
    }
}

File: src/main/java/com/practice/web/dto/UserDto.java
package com.practice.web.dto;

import com.practice.theater.models.User;

public class UserDto {

    private long id;
    private String email;
    private String phone;
    private String name;
    private String roleName;

    public static UserDto from(User user) {
        UserDto dto = new UserDto();
        dto.id = user.getId();
        dto.email = user.getEmail();
        dto.name = user.getName();
        dto.phone = user.getPhone();
        dto.roleName = user.getRole().name();
        return dto;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }
}
